zplugin-ros1/src/ros_to_zenoh_bridge/bridges_storage.rs

```diff
@@ -0,0 +1,272 @@
+//
+// Copyright (c) 2022 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use std::{collections::{HashMap, hash_map::Entry, HashSet}, sync::Arc};
+
+use super::{topic_bridge::{TopicBridge, self}, bridge_type::BridgeType, ros1_client, discovery::LocalResources, zenoh_client, topic_mapping::Ros1TopicMapping, ros1_to_zenoh_bridge_impl::BridgeStatus};
+
+
+struct Bridges {
+    publisher_bridges: HashMap<rosrust::api::Topic, TopicBridge>,
+    subscriber_bridges: HashMap<rosrust::api::Topic, TopicBridge>,
+    service_bridges: HashMap<rosrust::api::Topic, TopicBridge>,
+    client_bridges: HashMap<rosrust::api::Topic, TopicBridge>
+}
+impl Bridges {
+    fn new() -> Self {
+        Self { 
+            publisher_bridges: HashMap::new(),
+            subscriber_bridges: HashMap::new(),
+            service_bridges: HashMap::new(),
+            client_bridges: HashMap::new()
+        }
+    }
+
+    fn container_mut(&mut self, b_type: BridgeType) -> &mut HashMap<rosrust::api::Topic, TopicBridge> {
+        match b_type {
+            BridgeType::Publisher => &mut self.publisher_bridges,
+            BridgeType::Subscriber => &mut self.subscriber_bridges,
+            BridgeType::Service => &mut self.service_bridges,
+            BridgeType::Client => &mut self.client_bridges,
+        }
+    }
+
+    fn status(&self) -> BridgeStatus {
+        let fill = |status: &mut (usize, usize), bridges: &HashMap<rosrust::api::Topic, TopicBridge>| {
+            for (_topic, bridge) in bridges.iter() {
+                status.0 += 1;
+                if bridge.is_bridging() {
+                    status.1 += 1;
+                }
+            }
+        };
+
+        let mut result = BridgeStatus::default();
+        fill(&mut result.ros_clients, &self.client_bridges);
+        fill(&mut result.ros_publishers, &self.publisher_bridges);
+        fill(&mut result.ros_services, &self.service_bridges);
+        fill(&mut result.ros_subscribers, &self.subscriber_bridges);
+        return result;
+    }
+
+    fn clear(&mut self) {
```

Comment:
I suggest to either store `&'a ros1_client::Ros1Client` or `Arc<ros1_client::Ros1Client>`.